File: nodejsaliSpider/alifunc.js

function  retry_if_result_none(result){
    return result
}

function input_time_random(minNum,maxNum){
    switch(arguments.length){
            case 1:
                return parseInt(Math.random()*minNum+1,10);
                break;
            case 2:
                return parseInt(Math.random()*(maxNum-minNum+1)+minNum,10);
                break;
            default:
                return 0;
                break;
    }
}


async function getInfoByNumber(browser,mainPage,firstLevelCateroyNumber,pageNumber){
    page=mainPage;
    await page.waitFor(1000)   //延迟1秒
    try{
        console.log('开始获取分类信息...');
        page.click('.categories-content-title:first-child');
        // slider_again = await page.$eval('div#nc_1__scale_text > .nc-lang-cnt > b', node => node.textContent)
        // console.log("+++++++++++++++++++++++++++++++++slider_again")
        // console.log(slider_again)
        const mouse = page.mouse
	    // await mouse.move(rect.left, rect.top,{'delay': 1500})

    }
    catch(e){
        console.log(e, ':验证失败')
        return {"flag":false,"page":page}
    }
    await page.waitFor(1000)   //延迟1秒
    //
    // if(slider_again != 'Verified')
    //     return {"flag":false,"page":page}
    // else{
    //     await page.screenshot({'path': './headless-slide-result.png'})
    //     console.log('验证通过')
    //     return {"flag":true,"page":page}
    // }
}


module.exports = {
 retry_if_result_none,
 input_time_random,
 getInfoByNumber,
 // mouse_slide_getC
}

File: taobao-login-master/alifunc.py
from retrying import retry
import time, asyncio, random


def retry_if_result_none(result):
    return result is None


@retry(retry_on_result=retry_if_result_none,)
async def mouse_slide(page=None):
    await asyncio.sleep(3)
    try:
        print('开始验证...')
        # 鼠标移动到滑块，按下，滑动到头（然后延时处理），松开按键
        await page.hover('#nc_1_n1z') # 不同场景的验证码模块能名字不同。
        await page.mouse.down()

        await page.mouse.move(2000, 0, {'delay': random.randint(1000, 2000)})
        await page.mouse.up()
    except Exception as e:
        print(e, ':验证失败')
        return None
    else:
        await asyncio.sleep(3)
        # 判断是否通过
        slider_again = await page.Jeval('.nc-lang-cnt', 'node => node.textContent')
        if slider_again != '验证通过':
            return None
        else:
            await page.screenshot({'path': './headless-slide-result.png'})
            print('验证通过')
            return 1


def input_time_random():
    return random.randint(100, 151)
